pkg/transaction/eventgenerator.go

```diff
@@ -135,6 +136,14 @@ func (e *Generator) CreateEvents(summaryEvent LogEvent, detailEvents []LogEvent,
 		log.Debug("Found api path in traceability api exceptions list.  Ignore transaction event")
 		return events, nil
 	}
+	if util.IsNotTest() {
+		if summaryEvent.TransactionSummary != nil {
+			consumerDetails := e.getConsumerDetails(summaryEvent)
+			if consumerDetails != nil {
+				summaryEvent.TransactionSummary.ConsumerDetails = consumerDetails
```

Comment:
I think you can still set 

summaryEvent.TransactionSummary.ConsumerDetails = e.getConsumerDetails(summaryEvent)

If nil is returned then it will not be added to the event sent to condor with omitempty being here

ConsumerDetails *ConsumerDetails `json:"consumerDetails,omitempty"`